src/components/stack/BlockItem.js

```diff
@@ -12,21 +12,180 @@ class BlockItem extends Component {
 
     this.state = {
       hover: false,
-      modTitle: null,
-      modDescription: null,
-      modDurationWork: null,
-      modDurationBreak: null,
+      modTitle: null, //title of the task block
+      modDescription: null, //description of the task block
+      modDurationWork: null, //time length in seconds of burst
+      modDurationBreak: null, //time length in seconds of break
       focusState: FOCUS_NONE,
     };
+
+    this.handleOnHover = this.handleOnHover.bind(this);
+    this.handleOnNotHover = this.handleOnNotHover.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleClose = this.handleClose.bind(this);
```

Comment:
Prefer 'handleHover' over 'handleOnHover'. Prefer 'handleClickEdit' over 'handleEdit'. No need to use 'On', since it's kind of redundant. If the handler is responding to a button click, then use 'Click' in the name. You should probably rename 'handleOnHover' and 'handleOnNotHover' to 'handleMouseEnter' and 'HandleMouseLeave' I think. It makes more sense when considering what they're actually handling (onMouseEnter/Leave).

Reply:
Yea the 'On' that I was using is actually pretty redundant. Okay that makes sense, I'll incorporate the action that is triggering the event in the handler's name. Ah okay right, that makes more sense. I decided to rename them 'handleMouseEnter' and 'handleMouseLeave'. Yes it does.